drivers/video/video_stm32_dcmi.c

```diff
@@ -283,9 +408,12 @@ static int video_stm32_dcmi_set_stream(const struct device *dev, bool enable,
 static int video_stm32_dcmi_enqueue(const struct device *dev, struct video_buffer *vbuf)
 {
 	struct video_stm32_dcmi_data *data = dev->data;
-	const uint32_t buffer_size = data->fmt.pitch * data->fmt.height;
+	uint32_t buffer_size = data->fmt.pitch * data->fmt.height;
 
-	if (buffer_size > vbuf->size) {
+	/* Size of a framebuffer is unknown at the moment, assign maximum */
+	if (data->fmt.pixelformat == VIDEO_PIX_FMT_JPEG) {
+		buffer_size = vbuf->size;
+	} else if (buffer_size > vbuf->size) {
```

Comment:
I would be tempted to split the JPEG support in a different commit.

- The commit short subject does not talk about JPEG so harder to look at when it was introduced.
- Sometimes commits are reverted upstream to fix CI or locally troubleshoot. Then either JPEG or STM32U5 would be reverted but not both.